src/univariate/continuous/triangular.jl

```diff
@@ -119,27 +133,20 @@ function quantile(d::TriangularDist, p::Real)
               b - sqrt(b_m_a * (b - c) * (1 - p))
 end
 
-function mgf(d::TriangularDist{T}, t::Real) where T<:Real
-    if t == zero(t)
-        return one(T)
-    else
-        (a, b, c) = params(d)
-        u = (b - c) * exp(a * t) - (b - a) * exp(c * t) + (c - a) * exp(b * t)
-        v = (b - a) * (c - a) * (b - c) * t^2
-        return 2u / v
-    end
+function mgf(d::TriangularDist, t::Real)
+    a, b, c = params(d)
+    u = (b - c) * exp(a * t) - (b - a) * exp(c * t) + (c - a) * exp(b * t)
+    iszero(t) && return one(u)
+    v = (b - a) * (c - a) * (b - c) * t^2
+    return 2u / v
 end
 
-function cf(d::TriangularDist{T}, t::Real) where T<:Real
-    # Is this correct?
-    if t == zero(t)
-        return one(Complex{T})
-    else
-        (a, b, c) = params(d)
-        u = (b - c) * cis(a * t) - (b - a) * cis(c * t) + (c - a) * cis(b * t)
-        v = (b - a) * (c - a) * (b - c) * t^2
-        return -2u / v
-    end
+function cf(d::TriangularDist, t::Real)
+    a, b, c = params(d)
+    u = (b - c) * cis(a * t) - (b - a) * cis(c * t) + (c - a) * cis(b * t)
+    iszero(t) && return complex(one(u))
```

Comment:
I think this could be simplified to
```suggestion
    iszero(t) && return one(u)
```

Reply:
updated